be1-go/internal/popserver/handler/lao.go

```diff
@@ -0,0 +1,456 @@
+		storeMessage = false
+		errAnswer = handleRollCallClose(msg, channelPath)
+	case messagedata.RollCallObject + "#" + messagedata.RollCallActionCreate:
+		errAnswer = handleRollCallCreate(msg, channelPath)
+	case messagedata.RollCallObject + "#" + messagedata.RollCallActionOpen:
+		errAnswer = handleRollCallOpen(msg, channelPath)
+	case messagedata.RollCallObject + "#" + messagedata.RollCallActionReOpen:
+		errAnswer = handleRollCallReOpen(msg, channelPath)
+	case messagedata.ElectionObject + "#" + messagedata.ElectionActionSetup:
+		storeMessage = false
+		errAnswer = handleElectionSetup(msg, channelPath)
+	default:
+		errAnswer = answer.NewInvalidMessageFieldError("failed to handle %s#%s, invalid object#action", object, action)
+	}
+	if errAnswer != nil {
+		return errAnswer.Wrap("handleChannelLao")
+	}
+
+	if storeMessage {
+		db, errAnswer := database.GetLAORepositoryInstance()
+		if errAnswer != nil {
+			return errAnswer.Wrap("handleChannelLao")
+		}
+
+		err := db.StoreMessageAndData(channelPath, msg)
+		if err != nil {
+			errAnswer = answer.NewStoreDatabaseError("message: %v", err)
+			return errAnswer.Wrap("handleChannelLao")
+		}
+	}
+
+	errAnswer = broadcastToAllClients(msg, channelPath)
+	if errAnswer != nil {
+		return errAnswer.Wrap("handleChannelLao")
+	}
+	return nil
+}
+
+func handleRollCallCreate(msg message.Message, channelPath string) *answer.Error {
+	var rollCallCreate messagedata.RollCallCreate
+	errAnswer := msg.UnmarshalMsgData(&rollCallCreate)
+	if errAnswer != nil {
+		return errAnswer.Wrap("handleRollCallCreate")
+	}
+
+	errAnswer = rollCallCreate.Verify(channelPath)
+	if errAnswer != nil {
+		return errAnswer.Wrap("handleRollCallCreate")
+	}
+
+	return nil
+}
+
+func handleRollCallOpen(msg message.Message, channelPath string) *answer.Error {
+	var rollCallOpen messagedata.RollCallOpen
+	errAnswer := msg.UnmarshalMsgData(&rollCallOpen)
+	if errAnswer != nil {
+		return errAnswer.Wrap("handleRollCallOpen")
+	}
+
+	errAnswer = rollCallOpen.Verify(channelPath)
+	if errAnswer != nil {
+		return errAnswer.Wrap("handleRollCallOpen")
+	}
+
+	db, errAnswer := database.GetLAORepositoryInstance()
+	if errAnswer != nil {
+		return errAnswer.Wrap("handleRollCallOpen")
+	}
+
+	ok, err := db.CheckPrevID(channelPath, rollCallOpen.Opens, messagedata.RollCallActionCreate)
+	if err != nil {
+		errAnswer = answer.NewQueryDatabaseError("if previous id exists: %v", err)
+		return errAnswer.Wrap("handleRollCallOpen")
+	} else if !ok {
+		errAnswer = answer.NewInvalidMessageFieldError("previous id does not exist")
+		return errAnswer.Wrap("handleRollCallOpen")
+	}
+	return nil
+}
+
+func handleRollCallReOpen(msg message.Message, channelPath string) *answer.Error {
+	var rollCallReOpen messagedata.RollCallReOpen
+	errAnswer := msg.UnmarshalMsgData(&rollCallReOpen)
+	if errAnswer != nil {
+		return errAnswer.Wrap("handleRollCallReOpen")
+	}
+
+	errAnswer = handleRollCallOpen(msg, channelPath)
```

Comment:
Re open a rollcall doesn't work.
handleRollCallReOpen calls handleRollCallOpen, who calls CheckPrevID with the Opens field and a RollCallCreate

Reply:
it has been fixed